pkg/proxy/iptables/proxier.go

```diff
@@ -166,12 +167,26 @@ type endpointsInfo struct {
 	chainName utiliptables.Chain
 }
 
+// Returns just the IP part of an IP:port or IP endpoint string. If the IP
+// part is an IPv6 address enclosed in brackets (e.g. "[fd00:1::5]:9999"),
+// then the brackets are stripped as well.
+func ipPart(s string) string {
+	if index := strings.LastIndex(s, ":"); index != -1 {
+		ip := s[0:index]
+		// Strip off any surrounding brackets.
+		re := regexp.MustCompile(`\[(.*)\]`)
```

Comment:
Would a less compute intensive solution be to just check for a '[' at position zero and then take slice w/o first and last character?